docs/getting-started/advanced/docker-security.md

```diff
@@ -48,6 +48,9 @@ docker compose stop
 docker compose up -d
 ```
 
+Please keep in mind that the _numeric_ user and group ids are shared between the local docker host and the docker container (user/group _names_ are not). That means the user running inside the container can read and write all files on the host with the same user id as photoprism is running at. The container runtime tries restricts the available paths but as soon as a security vulnerability circumvents this docker imposed restrictions full access to the entire host with this user id is possible. Threfore nover run as root or with a low user id of regular users of the docker host.
```

Comment:
Could you please share an example of this security vulnerability?

Reply:
A good writeup to test/understand this user feature can be found here:
https://medium.com/@mccode/understanding-how-uid-and-gid-work-in-docker-containers-c37a01d01cf

A way to "use" this (and get root) is written here:
https://flast101.github.io/docker-privesc/

Even as the examples above use root / UID 0. it is the same for docker running with all user ids - the process inside the container can potentially access all files outside available to the same user id. Using (for example) UID 1000 or 1001 will give potential access to the files of the primary Linux user (as most Linux distributions start numbering with 1000 for regular users).

Using a high user id not used anywhere else on the docker host is a forward-looking security technique to not get owned if there is a security vulnerability found inside the Linux kernel or Docker daemon or some other software running inside the container. And all Docker security best practices mention a privilege escalation attack as possible scenario to take care of.

There is no real attack right now but you never now - better safe than sorrow..

e.g.
* https://www.oreilly.com/content/five-security-concerns-when-using-docker/
* https://snyk.io/de/blog/10-docker-image-security-best-practices/ -> 2. use least privileges and it is part of the Synk Docker image security checks
* https://docs.docker.com/engine/security/userns-remap/#enable-userns-remap-on-the-daemon -> prevent problem with user namespace remap
* https://www.tigera.io/learn/guides/container-security-best-practices/docker-security/
...

Reply:
just a note: restricting the UID to some specific values prevents Photoprism from running inside OpenShift Clusters as OpenShift automatically assigns a high random user id to every container started. I have not tested it right now but would assume its not working there...
https://www.redhat.com/en/blog/a-guide-to-openshift-and-uids